src/main/java/org/opensearch/security/action/apitokens/ApiTokenRepository.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.action.apitokens;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.security.authtoken.jwt.ExpiringBearerAuthToken;
+import org.opensearch.security.configuration.TokenListener;
+import org.opensearch.security.identity.SecurityTokenManager;
+import org.opensearch.security.securityconf.impl.v7.RoleV7;
+import org.opensearch.security.user.User;
+import org.opensearch.transport.client.Client;
+
+import static org.opensearch.security.http.ApiTokenAuthenticator.API_TOKEN_USER_PREFIX;
+
+public class ApiTokenRepository {
+    private final ApiTokenIndexHandler apiTokenIndexHandler;
+    private final SecurityTokenManager securityTokenManager;
+    private final List<TokenListener> tokenListener = new ArrayList<>();
+    private static final Logger log = LogManager.getLogger(ApiTokenRepository.class);
+
+    private final Map<String, RoleV7> jtis = new ConcurrentHashMap<>();
```

Comment:
What do you think about changing `RoleV7` into `SubjectBasedActionPrivileges` here? `PrivilegesEvaluator` could then retrieve these instances just from here.